### PR TITLE
Fix Service Manager DI. Change tests

### PR DESCRIPTION
**Pull request by marinovdf**

### CODE REVIEW

**Comment by stijnhau** [comment]
There is already a pr for this on the v1 branch.
THere still needs to be PR created for the master branche

File: src/ZfcUser/Authentication/Adapter/Db.php [hunk 1]
```diff
@@ -12,7 +12,7 @@
 use ZfcUser\Mapper\User as UserMapperInterface;
 use ZfcUser\Options\AuthenticationOptionsInterface;
 
-class Db extends AbstractAdapter implements ServiceManagerAwareInterface
```
Comment (svycka) [inline src/ZfcUser/Authentication/Adapter/Db.php]: removing `ServiceManagerAwareInterface` is BC break what if someone extended this and have their own factory but depend on initializers to add serviceManager.

File: src/ZfcUser/Authentication/Adapter/Db.php [hunk 2]
```diff
@@ -34,6 +34,11 @@ class Db extends AbstractAdapter implements ServiceManagerAwareInterface
      */
     protected $options;
 
+    public function __construct(ServiceManager $serviceManager)
```
Comment (svycka) [inline src/ZfcUser/Authentication/Adapter/Db.php]: for 1.* version better would be use setServiceLocator() method

File: src/ZfcUser/Controller/UserController.php [hunk 3]
```diff
@@ -348,7 +357,7 @@ public function changeEmailAction()
     public function getUserService()
     {
         if (!$this->userService) {
-            $this->userService = $this->getServiceLocator()->get('zfcuser_user_service');
```
Comment (svycka) [inline src/ZfcUser/Controller/UserController.php]: should not be changed if factory will use setServiceLocator()

File: src/ZfcUser/Authentication/Adapter/Db.php (hunk 2 above)
Comment (svycka) [inline src/ZfcUser/Authentication/Adapter/Db.php]: or can leave in constructor, but still use `$this->getServiceLocator()` and leave `ServiceManagerAwareInterface`

**Comment by svycka** [comment]
It's the same for all classes I haven't looked at tests but if they pass that's a good sign :D

File: src/ZfcUser/Controller/UserController.php [hunk 4]
```diff
@@ -61,14 +63,21 @@ class UserController extends AbstractActionController
...
+     * @var ServiceLocator
      */
-    public function __construct($redirectCallback)
+    protected $serviceLocator;
+
+    /**
+     * @param callable                      $redirectCallback
+     * @param ServiceManager $serviceManager
+     */
+    public function __construct($redirectCallback, ServiceManager $serviceManager)
```
Comment (svycka) [inline src/ZfcUser/Controller/UserController.php]: This is BC break for sure even for me as I extended this controller and have a `parent::__construct($redirectCallback)` call in my new constructor.  You should make $serviceManager optional or remove as all others constructors

**Comment by svycka** [comment]
@stijnhau master branch is for 2.0 and can contain BC break so this whole initializers thing can be removed. Maybe better create another a little bit different PR for master.

File: src/ZfcUser/Authentication/Adapter/Db.php (hunk 1 above)
Comment (weierophinney) [inline src/ZfcUser/Authentication/Adapter/Db.php]: @svycka Then it should be marked for a new minor release at the least, and potentially a new major release. However, making code _forwards_ compatible, and removing deprecation notices, is generally considered reasonable for a minor release. While you typically don't want to remove features, if doing so provides greater compatibility, it's the responsible thing to do.

**Comment by stijnhau** [comment]
@svycka 
Agreed but we already have https://github.com/ZF-Commons/ZfcUser/pull/627
for the v1
And nothing fot the master branch.
Ik looking at my planning to do iot myself.

**Comment by svycka** [comment]
@stijnhau but this PR is for 1.x branch
> @marinovdf  wants to merge 1 commit into ZF-Commons:1.x from marinovdf:sm_fix

**Comment by Danielss89** [comment]
Fixed in #637

**Pull request closed without merging**